research/context/preselection/preselect_wikipedia_comments.py

```diff
@@ -0,0 +1,125 @@
+# coding=utf-8
+# Copyright 2018 The Conversation-AI.github.io Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Preselect Wikipedia comments.
+
+A library to preselect wikipedia comments based on their toxicity score
+among others aspects.
+
+The initial goal was to create a sample of comments to use for annotation;
+the sample was stratified to ensure existence of toxic comments. If one
+uses 3 zones, then [0,1] will be divided in [0,0.33), [0.33,0.66) and
+[0.66,1], so that the comments are scored from all three zones.
+
+The BigQuery table queried to create this sample contains 240M rows (294Gb)
+and we assume that 'toxicity', 'cleaned_content', 'replyTo_id' and 'type'
+attributes exist.
```

Comment:
Good to say what the output looks like too :) 

e.g. that there is a new column for random number called "r", and also a new column for toxic_zone, where -1 indicates the random ramble, and i indicates toxicity in the range of `(i / 1, i+1 / 1]` or whatever it is exactly (presumably the start or end has to be [x,y]).